allennlp/tests/semparse/worlds/wikitables_variable_free_world_test.py

```diff
@@ -0,0 +1,206 @@
+
+        check_productions_match(valid_actions['<n,<n,<n,d>>>'],
+                                ['date'])
+
+        check_productions_match(valid_actions['<r,<d,<l,r>>>'],
+                                ['filter_date_equals', 'filter_date_greater',
+                                 'filter_date_greater_equals', 'filter_date_lesser',
+                                 'filter_date_lesser_equals', 'filter_date_not_equals'])
+
+        check_productions_match(valid_actions['<r,<l,n>>'],
+                                ['average', 'max', 'min', 'sum'])
+
+        check_productions_match(valid_actions['<r,r>'],
+                                ['first', 'last', 'next', 'previous'])
+
+        check_productions_match(valid_actions['<r,n>'],
+                                ['count'])
+
+        # These are the columns in table, and are instance specific.
+        check_productions_match(valid_actions['l'],
+                                ['fb:row.row.year',
+                                 'fb:row.row.league',
+                                 'fb:row.row.avg_attendance',
+                                 'fb:row.row.division',
+                                 'fb:row.row.regular_season',
+                                 'fb:row.row.playoffs',
+                                 'fb:row.row.open_cup'])
+
+        check_productions_match(valid_actions['@start@'],
+                                ['d', 'n', 's'])
+
+        # We merged cells and parts in SEMPRE to strings in this grammar.
+        check_productions_match(valid_actions['s'],
+                                ['fb:cell.2',
+                                 'fb:cell.2001',
+                                 'fb:cell.2005',
+                                 'fb:cell.4th_round',
+                                 'fb:cell.4th_western',
+                                 'fb:cell.5th',
+                                 'fb:cell.6_028',
+                                 'fb:cell.7_169',
+                                 'fb:cell.did_not_qualify',
+                                 'fb:cell.quarterfinals',
+                                 'fb:cell.usl_a_league',
+                                 'fb:cell.usl_first_division',
+                                 'fb:part.4th',
+                                 'fb:part.western',
+                                 'fb:part.5th',
+                                 '[<r,<l,s>>, r, l]'])
+
+        check_productions_match(valid_actions['d'],
+                                ['[<n,<n,<n,d>>>, n, n, n]'])
+
+        check_productions_match(valid_actions['n'],
+                                ['-1',
+                                 '0',
+                                 '1',
+                                 '2013',
+                                 '[<r,<l,n>>, r, l]',
+                                 '[<r,<r,<l,n>>>, r, r, l]',
+                                 '[<r,n>, r]'])
+
+        check_productions_match(valid_actions['r'],
```

Comment:
How does this language handle conjunctions or disjunctions?  I'd expect to see something like `r -> [<r,<r,r>>, r, r]`, with productions like "and" and "or".

Reply:
It doesn't (see Appendix A in the paper). I agree that this is limiting, and I'm curious to see how this affects the produced logical forms. Recall that our NLVR language does not have `and` and `or` too, which meant we could not translate certain kinds of sentences, but it made the language very simple.